### ISSUE TITLE
Request Libary Is Deprecated

### ISSUE THREAD

**Issue by settify-mattweston** [geckoboard/geckoboard-node#21]
Could you please update this package to no longer depend on request. It has been deprecated for a while and installing this node library causes vulnerability errors.

```
Severity: moderate
Server-Side Request Forgery in Request - https://github.com/advisories/GHSA-p8p7-x288-28g6
Depends on vulnerable versions of tough-cookie
```

See here for deprecated notice: 
https://www.npmjs.com/package/request
https://github.com/request/request

**Comment by t-o-m-** [comment]
@settify-mattweston thank you for opening this issue. We have just released a completely new version of this package - please see the current README for further details. If you're wiling to try it out, we're happy to receive any feedback.

**Comment by settify-mattweston** [comment]
@t-o-m-  I had a quick play with the revised library today. I am hitting an issue where one of the API requests is returning a `201` response when creating a dataset. This is for a `PUT /datasets/report.name` request.

I think this is being caused by the `res.status != 200` check [here](https://github.com/geckoboard/geckoboard-node/blob/d3952cd21620be2a9af2bb6c658884412cd6096a/src/index.ts#L391)

I'm assuming that `201` is a valid response and it should not be returning an error. As when I run the request the second time it would appear that the dataset was created.

**Comment by settify-mattweston** [comment]
@t-o-m- outside of the issue above the revisions are all working well. I have been able to create new datasets and append data as expected. Thanks for addressing the outdated library issue so quickly! 🙌

**Comment by glenjamin** [comment]
thanks for the feedback @settify-mattweston - we've just pushed a fix for the 201 issue in `2.0.2`
